src/Orleans/Streams/Core/StreamSubscriptionManager.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Orleans.Runtime;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Orleans.Streams.Core
+{
+    internal class StreamSubscriptionManager: IStreamSubscriptionManager
+    {
+        private IStreamPubSub streamPubSub;
+        private StreamPubSubType type;
+        private string streamProviderName;
+        //potential memory optimization on this
+        private ConcurrentDictionary<Guid, StreamSubscription> allSubscriptions;
```

Comment:
I don't think we need allSubscriptions.  PubSub system is authoritative on what subscriptions exist, so we shouldn't track this separately.